addons/sale_pdf_quote_builder/models/ir_actions_report.py

```diff
@@ -138,7 +138,7 @@ def _get_form_fields_mapping(self, order):
         }
 
         # Adding fields from each line, prefixed by the line_id to avoid conflicts
-        for line in order.order_line:
+        for line in order.order_line.filtered(lambda sol: sol.id in doc_line_id_mapping.values()):
```

Comment:
```suggestion
        lines_with_doc_ids = set(doc_line_id_mapping.values())
        for line in order.order_line.filtered(lambda sol: sol.id in lines_with_doc_ids):
```
Using a set here is even better, perf wise, as there could be multiple time the same id in the values =]